packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -975,16 +975,10 @@ class _CupertinoEdgeShadowPainter extends BoxPainter {
 
     final TextDirection? textDirection = configuration.textDirection;
     assert(textDirection != null);
-    final double start;
-    final double shadowDirection; // -1 for ltr, 1 for rtl.
```

Comment:
This comment was lost in the refactoring, but it is kinda useful. Can you preserve it?

Reply:
![shadowDirection](https://github.com/flutter/flutter/assets/10457200/313ac9c0-33c7-43cb-b567-8c1055b43ab4)

This is one of those borderline cases as far as whether a comment is useful: in my opinion, the `switch` expression makes it easy to tell that the value of `shadowDirection` is `1` for `rtl` and `-1` for `ltr`.